### DISCUSSION TITLE
How to call (or access) functions (or variable/objects) in a layout from a page using that layout?

### DISCUSSION THREAD

**Discussion by wureka** [nuxt/nuxt#24441] [Questions]
`my-layout.vue`
```
<script setup lang="ts">
defineExpose({
  showAlert
})
function showAlert() {
  console.log('showAlert() in dashboard')
}
</script>
<template>
  <div>
    <h1>MY-LAYOUT </h1>
    <slot />
  </div>
</template>
```

`parent.vue`

```
<script setup lang="ts">
const component = ref(null)
function testSlotAlert() {
  console.info("component:", component)
  component.value.showSlotAlert()
}

</script>
<template>
  <NuxtLayout name="my-layout"  ref="component">
    PARENT PAGE <v-btn @click="testSlotAlert"> TEST calling functions in layout  </v-btn>
  </NuxtLayout>
</template>

<style scoped>

</style>

```

After running, it happens below error:

`TypeError: component.value.showSlotAlert is not a function`

Could anyone suggest how to call (or access) functions (or variables) in layout ?

**Comment by xjccc** [comment]
```
// layout.vue

defineExpose({ showAlert })

// parent.vue

component.value.layoutRef.showAlert

```

**Reply by wureka** [reply]
I just gave it a shot, not work.
When  executing `component.value.showAlert`, no error, and nothing showed in console.
When executing `component.value.showAlert()`, it happened same error `TypeError: component.value.showSlotAlert is not a function`

**Reply by xjccc** [reply]
emm.. I'm log component.value, find it has a layoutRef and the expose value in it. or you can use defineEmits at layout, and @ at NuxtLayout?

**Reply by xjccc** [reply]
[NuxtLayout](https://nuxt.com/docs/api/components/nuxt-layout#layouts-ref)
